### PR TITLE
perf(query_subscriptions): Convert query subscription consumer to use multiprocessing

### PR DESCRIPTION
**Pull request by wedamija**
This converts the query subscription result consumer to use multiprocessing, using the same method as https://github.com/getsentry/sentry/pull/47037.

I've split it into separate files, so that the code used to build the consumer is separate from the actual code the consumer runs. Trying to avoid importing as much as possible before we fork and reconfigure Django.

### CODE REVIEW

**Comment by wedamija** [comment]
I'll test this out locally next week before merging

File: src/sentry/snuba/query_subscriptions/run.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    consumer = KafkaConsumer(
+        build_kafka_consumer_configuration(
+            cluster_options,
+            group_id=group_id,
+            strict_offset_reset=strict_offset_reset,
+            auto_offset_reset=initial_offset_reset,
+        )
+    )
+    metrics_wrapper = MetricsWrapper(metrics.backend, name="query_subscription_consumer")
+    configure_metrics(metrics_wrapper)
```
Comment (untitaker) [inline src/sentry/snuba/query_subscriptions/run.py:130]: you will need to do this in the subprocess as well. Take a look at https://github.com/getsentry/sentry/blob/f8c0324a0b0d49e110401468cc665d6c4cda0904/src/sentry/sentry_metrics/configuration.py#L169

we should (long-term) change arroyo to not make this necessary though, IMO

File: src/sentry/runner/commands/run.py [hunk 2]
```diff
@@ -461,17 +461,36 @@ def post_process_forwarder(**options):
     type=click.Choice(["earliest", "latest"]),
     help="Force subscriptions to start from a particular offset",
 )
+@kafka_options(
+    "occurrence-consumer",
```
Comment (ceorourke) [inline src/sentry/runner/commands/run.py:465]: Shouldn't this be query subscription consumer?

Comment (wedamija) [reply src/sentry/runner/commands/run.py:465]: That's what I get for copy pasting

**Comment by wedamija** [comment]
Ok, tested this locally and ironed out a few pickling related issues. I think we should be good to go now

**Review by ceorourke — APPROVED** [review]

**Pull request merged**